packages/zero-solid/src/solid-view.test.ts

```diff
@@ -1375,6 +1377,151 @@ test('basic with edit pushes', () => {
+  const row1Log: unknown[] = [];
+  const row0ALog: unknown[] = [];
+  const row0BLog: unknown[] = [];
+  const row1ALog: unknown[] = [];
+  const row1BLog: unknown[] = [];
+
+  function clearLog() {
+    row0Log.length = 0;
+    row1Log.length = 0;
+    row0ALog.length = 0;
+    row0BLog.length = 0;
+    row1ALog.length = 0;
+    row1BLog.length = 0;
+  }
+
+  new SolidView(
+    ms.connect([['a', 'asc']]),
+    onTransactionCommit,
+    {singular: false, relationships: {}},
+    () => {},
+    true,
+    () => {},
+    setState,
+  );
+
+  createEffect(() => {
+    row0Log.push(data()[0]);
+  });
+  createEffect(() => {
+    row1Log.push(data()[1]);
+  });
+  createEffect(() => {
+    row0ALog.push(data()[0]?.a);
+  });
+  createEffect(() => {
+    row0BLog.push(data()[0]?.b);
+  });
+  createEffect(() => {
+    row1ALog.push(data()[1]?.a);
+  });
+  createEffect(() => {
+    row1BLog.push(data()[1]?.b);
+  });
+
+  const data0 = [
+    {a: 1, b: 'a', [refCountSymbol]: 1, [idSymbol]: '1'},
+    {a: 2, b: 'b', [refCountSymbol]: 1, [idSymbol]: '2'},
+  ];
+  expect(data()).toEqual(data0);
+  expect(row0Log).toEqual([
+    {a: 1, b: 'a', [refCountSymbol]: 1, [idSymbol]: '1'},
+  ]);
+  expect(row1Log).toEqual([
+    {a: 2, b: 'b', [refCountSymbol]: 1, [idSymbol]: '2'},
+  ]);
+  expect(row0ALog).toEqual([1]);
+  expect(row0BLog).toEqual(['a']);
+  expect(row1ALog).toEqual([2]);
+  expect(row1BLog).toEqual(['b']);
+
+  clearLog();
+
+  ms.push({type: 'edit', row: {a: 2, b: 'b2'}, oldRow: {a: 2, b: 'b'}});
+
+  expect(data()).toEqual(data0);
+  commit();
+
+  const data1 = [
+    {a: 1, b: 'a', [refCountSymbol]: 1, [idSymbol]: '1'},
+    {a: 2, b: 'b2', [refCountSymbol]: 1, [idSymbol]: '2'},
+  ];
+  expect(data()).toEqual(data1);
+  expect(row0Log).toEqual([]);
+  expect(row1Log).toEqual([]);
+  expect(row0ALog).toEqual([]);
+  expect(row0BLog).toEqual([]);
+  expect(row1ALog).toEqual([]);
+  expect(row1BLog).toEqual(['b2']);
+
+  clearLog();
+
+  ms.push({type: 'edit', row: {a: 3, b: 'b3'}, oldRow: {a: 2, b: 'b2'}});
+
+  expect(data()).toEqual(data1);
+  commit();
+  const data2 = [
+    {a: 1, b: 'a', [refCountSymbol]: 1, [idSymbol]: '1'},
+    {a: 3, b: 'b3', [refCountSymbol]: 1, [idSymbol]: '3'},
+  ];
+  expect(data()).toEqual(data2);
+  expect(row0Log).toEqual([]);
+  expect(row1Log).toEqual([]);
+  expect(row0ALog).toEqual([]);
+  expect(row0BLog).toEqual([]);
+  expect(row1ALog).toEqual([3]);
+  expect(row1BLog).toEqual(['b3']);
+
+  clearLog();
+
+  ms.push({type: 'edit', row: {a: 0, b: 'b3'}, oldRow: {a: 3, b: 'b3'}});
+
+  expect(data()).toEqual(data2);
+  commit();
+  const data3 = [
+    {a: 0, b: 'b3', [refCountSymbol]: 1, [idSymbol]: '0'},
+    {a: 1, b: 'a', [refCountSymbol]: 1, [idSymbol]: '1'},
+  ];
+  expect(data()).toEqual(data3);
+  expect(row0Log).toEqual([
+    {a: 0, b: 'b3', [refCountSymbol]: 1, [idSymbol]: '0'},
+  ]);
+  expect(row1Log).toEqual([
+    {a: 1, b: 'a', [refCountSymbol]: 1, [idSymbol]: '1'},
```

Comment:
Why did this change?

Reply:
Here I effectively swapped the two rows ({a: 3, b: 'b3'}, changed to {a: 0, b: 'b3'} and moved to index 0, and {a: 1, b: 'a'}, move to index1 ), so it makes sense that the effects reading these two indexes would fire.